src/components/DAOPicker.js

```diff
@@ -0,0 +1,171 @@
+          DAOApi.createFilter(searchBy).contains(
+            this._searchTextStore.debouncedText,
+          ),
+        ],
+      });
+
+      this._listStore.reload();
+    });
+  }
+
+  _listKeyExtractor = (row: Row<TEntity>): string => row.key;
+
+  _renderRow = (
+    renderRowProps: RenderRowProps<TEntity>,
+  ): React.Element<any> => {
+    const { checkIsSelected, toggleItem } = this.injectedProps;
+    const { renderRow } = this.props;
+    const { item: { loader } } = renderRowProps;
+    const isSelected =
+      loader.hasValue() && checkIsSelected(loader.getValueEnforcing());
+
+    return renderRow(({ ...renderRowProps, isSelected, toggleItem }: any));
+  };
+
+  render() {
+    const {
+      headerTitle,
+      label,
+      multiple,
+      placeholder,
+      stringValueExtractor,
+    } = this.props;
+    const { clear, value } = this.injectedProps;
+
+    return (
+      <View>
+        <PickerTextInput
+          label={label}
+          onPress={this._modalToggleStore.toggleOn}
+          placeholder={placeholder}
+          stringValueExtractor={stringValueExtractor}
+          value={value}
+        />
+        <Modal isVisible={this._modalToggleStore.isToggled}>
+          {this._modalToggleStore.isToggled && (
+            <Container>
+              <Header
+                leftComponent={
+                  <HeaderIconButton
+                    name="arrow-back"
+                    onPress={this._modalToggleStore.toggleOff}
+                  />
+                }
+                rightComponent={
+                  <HeaderIconButton
+                    name="search"
+                    onPress={this._searchToggleStore.toggle}
+                  />
+                }
+                title={headerTitle}
+              />
+              {this._searchToggleStore.isToggled && (
+                <SearchBar
+                  onChangeText={this._searchTextStore.setText}
+                  showLoadingIcon={this._listStore.isFetchingRemoteCount}
+                  value={this._searchTextStore.text}
+                />
+              )}
+              <List
+                data={this._listStore.rows}
+                extraData={Array.isArray(value) ? value.length : value}
+                keyExtractor={this._listKeyExtractor}
+                ListFooterComponent={
+                  <LoadingListFooter
+                    isLoading={this._listStore.isFetchingRemoteCount}
+                  />
+                }
+                onEndReached={this._listStore.fetchNextPage}
+                onRefresh={this._listStore.reload}
+                renderItem={this._renderRow}
+              />
+              <PickerControl
+                multiple={multiple}
+                onClearPress={clear}
+                onSelectPress={this._modalToggleStore.toggleOff}
+                value={value}
+              />
+            </Container>
+          )}
+        </Modal>
+      </View>
+    );
+  }
+}
+
+export default SearchPicker;
```

Comment:
The whole file looks magic. Let me know if you want me to check out specific patterns.

Reply:
Heh... I have a hard time with generic types in withPicker.js + pickerStore.js ...But I don't think that you can help me with this in a review. Other than that I think it works pretty well.